app/services/employment/CheckEmploymentDetailsService.scala

```diff
@@ -0,0 +1,102 @@
+import uk.gov.hmrc.http.HeaderCarrier
+import uk.gov.hmrc.play.audit.http.connector.AuditResult
+
+import javax.inject.Inject
+import scala.concurrent.{ExecutionContext, Future}
+
+class CheckEmploymentDetailsService @Inject()(employmentSessionService: EmploymentSessionService,
+                                              nrsService: NrsService,
+                                              auditService: AuditService) {
+
+  def performSubmitAudits(model: CreateUpdateEmploymentRequest, employmentId: String, taxYear: Int, prior: Option[AllEmploymentData])
+                         (implicit user: User[_], hc: HeaderCarrier, ec: ExecutionContext): Future[AuditResult] = {
+
+    val audit: Either[AuditModel[AmendEmploymentDetailsUpdateAudit], AuditModel[CreateNewEmploymentDetailsAudit]] = prior.flatMap {
+      prior =>
+        val priorData = employmentSessionService.employmentSourceToUse(prior, employmentId, isInYear = false)
+        priorData.map(prior => model.toAmendAuditModel(employmentId, taxYear, prior._1).toAuditModel)
+    }.map(Left(_)).getOrElse {
+
+      val existingEmployments = prior.map {
+        prior =>
+          employmentSessionService.getLatestEmploymentData(prior, isInYear = false).map {
+            employment =>
+              PriorEmploymentAuditInfo(employment.employerName, employment.employerRef)
+          }
+      }.getOrElse(Seq.empty)
+
+      Right(model.toCreateAuditModel(taxYear, existingEmployments = existingEmployments).toAuditModel)
+    }
+
+    audit match {
+      case Left(amend) => auditService.sendAudit(amend)
+      case Right(create) => auditService.sendAudit(create)
+    }
+  }
+
+  def performSubmitNrsPayload(model: CreateUpdateEmploymentRequest, employmentId: String, prior: Option[AllEmploymentData])
+                             (implicit user: User[_], request: Request[_], hc: HeaderCarrier): Future[NrsSubmissionResponse] = {
+    val nrsPayload: Either[DecodedAmendEmploymentDetailsPayload, DecodedCreateNewEmploymentDetailsPayload] = prior.flatMap {
+      prior =>
+        val priorData = employmentSessionService.employmentSourceToUse(prior, employmentId, isInYear = false)
+        priorData.map(prior => model.toAmendDecodedPayloadModel(employmentId, prior._1))
+    }.map(Left(_)).getOrElse {
+
+      val existingEmployments = prior.map {
+        prior =>
+          employmentSessionService.getLatestEmploymentData(prior, isInYear = false).map {
+            employment =>
+              DecodedPriorEmploymentInfo(employment.employerName, employment.employerRef)
+          }
+      }.getOrElse(Seq.empty)
+
+      Right(model.toCreateDecodedPayloadModel(existingEmployments))
+    }
+
+    nrsPayload match {
+      case Left(amend) => nrsService.submit(user.nino, amend, user.mtditid)
+      case Right(create) => nrsService.submit(user.nino, create, user.mtditid)
+    }
+  }
+
+  def isSingleEmploymentAndAudit(employmentDetails: EmploymentDetailsViewModel,
```

Comment:
Method name is duplicated in CheckYourBenefitsService. They are different methods with different inputs so I think it's fine, but it could cause some minor confusion in future.

Reply:
They also come from different services, so you need to specify the service name too. This is fine. We have same methods (show and submit) in our Controllers for example but they do not cause confusion I suppose.
To be fair, I only moved this method from the controller. Still not ideal though. I would push for further refactoring in the future :)

Reply:
Yeah I think that's pretty reasonable. I think it should be ok as well, but it's worth thinking about. I'll approve for now and maybe we'll do something about it in a further refactoring